File: core/ErChaShuDeJingXiangLcof/ErChaShuDeJingXiangLcof.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


def mirrorTree(root: TreeNode) -> TreeNode:
    if root is None:
        return root

    left_node = mirrorTree(root.left)
    right_node = mirrorTree(root.right)
    root.left, root.right = right_node, left_node
    return root

File: core/ErChaShuDeJingXiangLcof/ErChaShuDeJingXiangLcof.java
package com.dbc.code;

public class ErChaShuDeJingXiangLcof {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

    public TreeNode mirrorTree(TreeNode root) {
        if (root == null) return root;
        TreeNode leftNode = mirrorTree(root.left);
        TreeNode rightNode = mirrorTree(root.right);
        root.left = rightNode;
        root.right = leftNode;
        return root;
    }
}
